class.jetpack-onboarding-end-points.php

```diff
@@ -531,12 +533,54 @@ static function install_theme() {
 					wp_send_json_error('Could not install theme (unspecified server error)');
 					die();
 				}
-			} 
+			}
 		}
 
 		wp_send_json_success( $theme_id );
 	}
 
+	static function have_contact_info_widget( $widgets ) {
+		foreach ($widgets as $widget) {
+    		if ( strpos( $widget, 'widget_widget_contact_info' ) ) {
+    			return true;
+    		}
+    	}
+    	return false;
+	}
+
+	static function add_business_address() {
+		check_ajax_referer( self::AJAX_NONCE, 'nonce' );
+		$active_sidebars = get_option( 'sidebars_widgets' );
+		$widget_options = get_option( 'widget_widget_contact_info' );
+		if( isset( $active_sidebars['sidebar-1'] ) &&
+			( ! isset( $widget_options[1] ) || str_replace(array("\n", "\r"), '', $widget_options[1]['address'] )== '3999 Mission Boulevard, San Diego CA 92109' ) // default address
+			) {
+			$title =  wp_unslash( $_REQUEST['business_name'] );
+			$address = wp_unslash( $_REQUEST['business_address_1'] . ' ' .
+				$_REQUEST['business_address_2']. ' ' .
+				$_REQUEST['business_city']. ' ' .
+				$_REQUEST['business_state']. ' ' .
+				$_REQUEST['business_zip']. ' ' );
+			$phone = wp_unslash( $_REQUEST['phone'] );
+			$hours = wp_unslash( $_REQUEST['hours'] );
+			$widget_options[1] = array( 'title' => $title, 'address' => $address, 'phone' => $phone, 'hours' => $hours, 'showmap' => false );
+			$widget_options[0] = 1;
+			update_option( 'widget_widget_contact_info', $widget_options ); //update widget default options
+			if ( ! self::have_contact_info_widget( $active_sidebars['sidebar-1'] ) ) {
```

Comment:
If we instead ask the user if they wish to insert the widget, such logic will not be necessary.